mesh/core/src/main/protobuf/resolver.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+
+package io.linkerd.mesh;
+
+import "dtab.proto";
+
+/**
+ * Supports resolution of a concrete name to a replica set.
+ *
+ * Status: Experimental; this API is not yet finalized.
+ */
+service Resolver {
+
+  rpc GetReplicas (ReplicasReq) returns (Replicas) {}
+  rpc StreamReplicas (ReplicasReq) returns (stream Replicas) {}
+}
+
+message ReplicasReq {
+  /**
+   * A concrete cluster id.
+   */
+  Path id = 1;
+}
+
+/**
+ * Describes a TCP endpoint.
+ */
+message Endpoint {
```

Comment:
Here is how this updated terminology would affect our current nomenclature.

![naming-v3](https://cloud.githubusercontent.com/assets/3979810/23043990/2b73db82-f453-11e6-9cc6-6c2993fa51ce.png)